src/components/DropDownMenu.tsx

```diff
@@ -0,0 +1,256 @@
+import { Fragment, ReactNode } from "react"
+import { Menu, MenuItemsProps, Transition } from "@headlessui/react"
+
+import Icon, { IconProps } from "./Icons/Icon"
+import SignInButton from "./SignInButton"
+import SignOutButton from "./SignOutButton"
+import ThemeButton from "./ThemeButton"
+
+import { cl } from "../utils"
+
+import { Link } from "react-router-dom"
+import { isEmpty } from "lodash-es"
+
+import { useIsAdmin } from "../hooks/useIsAdmin"
+import { useServiceMessages } from "../api/requests"
+
+import ServiceMessageBadge from "./ServiceMessageBadge"
+import ExternalLink from "./ExternalLink"
+
+import { ReactComponent as KnowitLogo } from "/assets/svg/Knowit logo.svg"
+
+
+export const LinkButton = ({
+  to,
+  name,
+  content,
+  external = false
+}: {
+  to: string
+  name?: IconProps["name"]
+  content: ReactNode
+  external?: boolean
+}) => (
+  <Link
+    to={to}
+    rel={external ? "noopener noreferrer" : ""}
+    target={external ? "_blank" : "_self"}
+  >
+    <div
+      className={cl(
+        `
+      flex
+      flex-row
+      items-center
+      gap-3
+
+      whitespace-nowrap
+      rounded-full
+      bg-purple-700
+      px-12
+
+      py-3
+      hover:bg-purple-900
+      hover:ring
+      hover:ring-inset
+
+      hover:ring-purple-700
+      focus:outline-none
+      focus:ring
+      focus:ring-inset
+
+      focus:ring-purple-100
+      active:bg-purple-500
+`
+      )}
+    >
+      {name && <Icon name={name} standalone className="h-10 w-10" />}
+      {content}
+    </div>
+  </Link>
+)
+
+const DropDownMenuItem = ({
+  children,
+  ...rest
+}: MenuItemsProps<"div"> & { children: ReactNode }) => {
+  return (
+    <Menu.Item>
+      {({ active, close }) => (
+        <div
+          {...rest}
+          onClick={close}
+          className={cl(
+            active &&
+              "[&>button]:ring [&>button]:ring-inset [&_div]:ring [&_div]:ring-inset "
```

Comment:
Oof. Bruke groups heller? https://tailwindcss.com/docs/hover-focus-and-other-states#styling-based-on-parent-state

Reply:
Oof no more g, fikset